explorer/ast/member.h

```diff
@@ -25,6 +25,16 @@ struct NamedValue {
   Nonnull<const Value*> value;
 };
 
+// A IndexedValue represents a value identified by an index, such as a tuple
+// field
+struct IndexedValue {
+  // The field index.
+  size_t index;
+
+  // The field's value.
+  Nonnull<const Value*> value;
+};
```

Comment:
Looking at how other code handles this, I see NamedValue used:
https://github.com/carbon-language/carbon-lang/blob/trunk/explorer/interpreter/interpreter.cpp#L468-L473

That makes this feel a bit like it's adding a new way to index tuples; can we instead do it with NamedValue per the referenced code? I believe then member also wouldn't be changed, since NamedValue is already supported.

(note, I'm omitting a few comments specific to IndexedValue assuming it'll be removed)